<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <!-- Meta tags for social media banners, these should be filled in appropriatly as they are your "business card" -->
    <!-- Replace the content tag with appropriate information -->
    <meta name="description"
          content=" We propose GenomAgent, a multi-agent framework designed to advance genomic Question Answering by overcoming the rigidity and scalability limitations of single-agent systems.">
    <meta property="og:title"
          content="When Agents Get Lost: Dissecting Failure Modes in Graph-Based Navigation Instruction Evaluation"/>
    <meta property="og:description"
          content=" We propose GenomAgent, a multi-agent framework designed to advance genomic Question Answering by overcoming the rigidity and scalability limitations of single-agent systems."/>
    <meta property="og:url" content="https://kimia-abedini.github.io/Genom-Agent/"/>
    <!-- Path to banner image, should be in the path listed below. Optimal dimenssions are 1200X630-->
    <meta property="og:image" content="static/image/GenomAgent.png"/>
    <meta property="og:image:width" content="1200"/>
    <meta property="og:image:height" content="500"/>


    <meta name="twitter:title"
          content="When Agents Get Lost: Dissecting Failure Modes in Graph-Based Navigation Instruction Evaluation">
    <meta name="twitter:description"
          content=" We propose GenomAgent, a multi-agent framework designed to advance genomic Question Answering by overcoming the rigidity and scalability limitations of single-agent systems.">
    <!-- Path to banner image, should be in the path listed below. Optimal dimenssions are 1200X600-->
    <meta name="twitter:image" content="static/images/GenomAgent.png">
    <meta name="twitter:card" content="summary_large_image">
    <!-- Keywords for your paper to be indexed by-->
    <meta name="keywords"
          content="GeneGPT, Genomic Question Answering, Large Language Models, GeneTuring, Multi-Agent Systems, Bioinformatics, Genomics, AI in Healthcare, Biomedical Research">
    <meta name="viewport" content="width=device-width, initial-scale=1">


    <title>When Agents Get Lost: Dissecting Failure Modes in Graph-Based Navigation Instruction Evaluation</title>
    <link rel="icon" type="image/x-icon" href="static/images/favicon.ico">
    <link href="https://fonts.googleapis.com/css?family=Google+Sans|Noto+Sans|Castoro"
          rel="stylesheet">

    <link rel="stylesheet" href="static/css/bulma.min.css">
    <link rel="stylesheet" href="static/css/bulma-carousel.min.css">
    <link rel="stylesheet" href="static/css/bulma-slider.min.css">
    <link rel="stylesheet" href="static/css/fontawesome.all.min.css">
    <link rel="stylesheet"
          href="https://cdn.jsdelivr.net/gh/jpswalsh/academicons@1/css/academicons.min.css">
    <link rel="stylesheet" href="static/css/index.css">

    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
    <script src="https://documentcloud.adobe.com/view-sdk/main.js"></script>
    <script defer src="static/js/fontawesome.all.min.js"></script>
    <script src="static/js/bulma-carousel.min.js"></script>
    <script src="static/js/bulma-slider.min.js"></script>
    <script src="static/js/index.js"></script>
    <script type="text/javascript" async
            src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.7/MathJax.js?config=TeX-MML-AM_CHTML"></script>
</head>
<body>


<section class="hero">
    <div class="hero-body">
        <div class="container is-max-desktop">
            <div class="columns is-centered">
                <div class="column has-text-centered">
                    <h1 class="title is-1 publication-title">When Agents Get Lost: Dissecting Failure Modes in Graph-Based Navigation Instruction Evaluation</h1>
                    <div class="is-size-5 publication-authors">
                        <!-- Paper authors -->
                        <span class="author-block">
                            <a href="mailto:farzad.shami@aalto.fi" target="_blank">Farzad Shami<sup>ðŸš²</sup></a>,</span>
                        <span class="author-block">
                            <a href="mailto:kimia1999abedini@gmail.com" target="_blank">Kimia Abedini<sup>ðŸ›´</sup></a>,</span>
                        <span class="author-block">
                            <a href="mailto:seyed.h.hosseini@aalto.fi" target="_blank">Seyed Hosein Hoesini<sup>ðŸš²</sup></a>,</span>
                        <span class="author-block">
                            <a href="mailto:henrikki.tenkanen@aalto.fi" target="_blank">Henrikki Tenkanen<sup>ðŸš²</sup></a></span>
                    </div>

                    <div class="is-size-5 publication-authors">
                        <span class="author-block">Aalto University (ðŸš² Built Environment; ðŸ›´Computer Science)</span>
                    </div>

                    <div class="column has-text-centered">
                        <div class="publication-links">
                            <!-- Arxiv PDF link -->
                            <span class="link-block">
                                <a href="https://arxiv.org/abs/2601.10581" target="_blank" class="external-link button is-normal is-rounded is-dark">
                                    <span class="icon">
                                        <i class="fas fa-file-pdf"></i>
                                    </span>
                                    <span>Paper</span>
                                </a>
                            </span>
                            <!-- Github link -->
                            <span class="link-block">
                                <a href="https://github.com/fuzsh/lost" target="_blank" class="external-link button is-normal is-rounded is-dark">
                                    <span class="icon">
                                        <i class="fab fa-github"></i>
                                    </span>
                                    <span>Code</span>
                                </a>
                            </span>
                            <span class="link-block">
                                <a href="https://fuzsh.github.io/lost/explorer-tool" target="_blank" class="external-link button is-normal is-rounded is-dark">
                                    <span class="icon">
                                        <i class="fa-light fa-eye"></i>
                                    </span>
                                    <span>Data Explore</span>
                                </a>
                            </span>

                            <span class="link-block">
                                <a href="https://fuzsh.github.io/lost/annotation-tool" target="_blank" class="external-link button is-normal is-rounded is-dark">
                                    <span class="icon">
                                        <i class="fa-light fa-pencil"></i>
                                    </span>
                                    <span>Annotation</span>
                                </a>
                            </span>

                        </div>
                    </div>
                </div>
            </div>
        </div>
    </div>
</section>

<!-- Teaser Image-->
<section class="hero teaser">
    <div class="container is-max-desktop">
        <div class="hero-body">
            <img src="static/images/GenomAgent.png" alt="An overview of our method"/>
            <h2 class="subtitle has-text-centered">
                We propose GenomAgent, a multi-agent framework designed to advance genomic Question Answering by
                overcoming the rigidity and scalability limitations of single-agent systems.
                We achieve superior accuracy in complex tasks like sequence alignment while significantly reducing
                computational costs compared to existing state-of-the-art methods.
            </h2>
        </div>
    </div>
</section>
<!-- End teaser Image -->


<!-- Paper abstract -->
<section class="section hero is-light">
    <div class="container is-max-desktop">
        <div class="columns is-centered has-text-centered">
            <div class="column is-four-fifths">
                <h2 class="title is-3">Abstract</h2>
                <div class="content has-text-justified">
                    <p>
                        Comprehending genomic information is essential for biomedical research, yet extracting data from
                        complex distributed databases remains challenging.
                        Large language models (LLMs) offer potential for genomic Question Answering (QA) but face
                        limitations due to restricted access to domain-specific databases.
                        GeneGPT is the current state-of-the-art system that enhances LLMs by utilizing specialized API
                        calls, though it is constrained by rigid API dependencies and limited adaptability.
                        We replicate GeneGPT and propose GenomAgent, a multi-agent framework that efficiently
                        coordinates specialized agents for complex genomics queries.
                        Evaluated on nine tasks from the GeneTuring benchmark, GenomAgent outperforms GeneGPT by 12% on
                        average, and its flexible architecture extends beyond genomics to various scientific domains
                        needing expert knowledge extraction.
                    </p>
                </div>
            </div>
        </div>
    </div>
</section>
<!-- End paper abstract -->


<section class="section">
    <div class="container is-max-desktop">
        <div class="columns is-vcentered">
            <!-- Text Section (Left) -->
            <div class="column is-full">
                <div class="content">
                    <h2 class="title is-3">Task & Motivation</h2>
                    <p>
                        The objective is to enable LLMs to comprehend and extract expert knowledge from complex,
                        distributed genomic databases to answer natural language queries.
                    </p>
                    <p>
                        The work motivated by several limitations inherent in standard LLMs and the current
                        state-of-the-art system, <i>GeneGPT</i>:
                    </p>
                    <p>
                        1. <strong>Complexity of Genomic Data:</strong> Extracting data from distributed biomedical
                        databases remains a significant challenge for researchers. Standard LLMs struggle with this
                        because they have restricted access to domain-specific databases.<br>
                        2. <strong>Fragility of Existing Systems (GeneGPT):</strong> While GeneGPT is effective, it
                        relies on a single-agent architecture with rigid dependencies on specific API formats. This
                        makes the system fragile when interfacing with evolving tools.<br>
                        3. <strong>Context and Focus Issues:</strong> GeneGPT relies on extensive context windows, which
                        can lead to "attention dilution" where the model loses focus on the original query.
                    </p>
                </div>
            </div>
        </div>
    </div>
</section>


<section class="section hero is-small is-light">
    <div class="container is-max-desktop">
        <div class="columns is-centered">
            <div class="column is-full">
                <div class="content">
                    <h2 class="title is-3">Experiments</h2>
                    <div class="level-set has-text-justified">
                        <p>
                        <p>
                            Here, we report the performance and cost of <strong>GenomAgent</strong> in GeneTuring tasks
                            compared to GeneGPT's main results.
                            <strong>GenomAgent</strong> achieves substantial improvements in both performance and
                            computational efficiency.
                            Our model attains an average score of 0.93, exceeding the best-performing GeneGPT model
                            (0.83).
                        </p>

                        <p>
                            In simple tasks (nomenclature and genomic location), our system achieves near-perfect
                            performance with a score of 0.98, surpassing GeneGPT-slim's scores of 0.92 for nomenclature
                            and 0.88 for genomic location.
                            Most notably, in alignment tasks, which are the most challenging task for GeneGPT, we
                            achieve a remarkable 28.8% improvement.
                        </p>

                        <p>
                            Computational cost analysis reveals even more striking improvements.
                            <strong>GenomAgent</strong> costs only $2.11 total in all tasks (79.0% reduction from
                            best-performing GeneGPT ($10.06)).
                        </p>
                    </div>
                </div>
            </div>
        </div>

        <div class="hero-body">
            <img src="static/images/results.png" alt="A figure showing our ProbeLog logit descriptors"/>
        </div>

    </div>
</section>


<section class="section hero is-small">
    <div class="container is-max-desktop">
        <div class="columns is-centered">
            <div class="column is-full">
                <div class="content">
                    <h2 class="title is-3">Performance-cost tradeoff on GeneTuring.</h2>
                    <div class="level-set has-text-justified">
                        <p>
                            In addition, we show <strong>GenomAgent</strong> is the optimal selection, as it achieves a
                            high score at minimal computational expense.
                            Bubble size shows normalized cost and High Value Region shows optimal performance at minimal
                            cost.
                        </p>
                    </div>
                </div>
            </div>
        </div>

        <div class="hero-body">
            <img src="static/images/trade_off.png"
                 alt="A figure showing our zero-shot ProbeLog logit descriptors"
                 style="display: block; margin-left: auto; margin-right: auto; width: 75%; height: auto;"
            />
        </div>

    </div>
</section>

<!--BibTex citation -->
<section class="section" id="BibTeX">
    <div class="container is-max-desktop content">
        <h2 class="title">BibTeX</h2>
        <pre><code>@misc{.....,
      title={When Agents Get Lost: Dissecting Failure Modes in Graph-Based Navigation Instruction Evaluation}, 
      author={......},
      year={2026},
      eprint={2601.10581},
      archivePrefix={arXiv},
      primaryClass={cs.AI},
      url={.......},
}</code></pre>
    </div>
</section>
<!--End BibTex citation -->


<footer class="footer">
    <div class="container">
        <div class="columns is-centered">
            <div class="column is-8">
                <div class="content">

                    <p>
                        This website is adapted from <a href="https://nerfies.github.io" target="_blank">Nerfies</a>,
                        licensed under a <a rel="license"
                                            href="http://creativecommons.org/licenses/by-sa/4.0/"
                                            target="_blank">Creative Commons Attribution-ShareAlike 4.0 International
                        License</a>.
                    </p>

                </div>
            </div>
        </div>
    </div>
</footer>


</body>
</html>
